corporate/views/remote_billing_page.py

```diff
@@ -0,0 +1,192 @@
+import logging
+from typing import Optional, TypedDict
+
+from django.conf import settings
+from django.core import signing
+from django.http import HttpRequest, HttpResponse, HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import csrf_exempt
+from pydantic import Json
+
+from zerver.decorator import self_hosting_management_endpoint
+from zerver.lib.exceptions import JsonableError, MissingRemoteRealmError
+from zerver.lib.remote_server import RealmDataForAnalytics, UserDataForRemoteBilling
+from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import PathOnly, typed_endpoint
+from zilencer.models import RemoteRealm, RemoteZulipServer
+
+billing_logger = logging.getLogger("corporate.stripe")
+
+
+class RemoteBillingIdentityDict(TypedDict):
+    user_uuid: str
+    user_email: str
+    user_full_name: str
+    remote_server_uuid: str
+    remote_realm_uuid: str
+
+
+@csrf_exempt
+@typed_endpoint
+def remote_server_billing_entry(
```

Comment:
Should this be decorated with  `@self_hosting_management_endpoint`?

Reply:
As mentioned above, I didn't put the decorator, because this is a bouncer endpoint - it lives in `zilencer/urls.py` and is queried by the remote server with its bouncer credentials. So this doesn't get queried on the `selfhosting.` subdomain.

I don't think we would want this on  `selfhosting.`  at all, because then server would need to have that subdomain hard-coded in their settings or something. So it seems right to me for this to be requested at `PUSH_NOTIFICATION_BOUNCER_URL`.

Reply:
OK, that makes sense to me. Though maybe we should maybe a parallel `PUSH_BOUNCER_SUBDOMAIN` setting `@push_bouncer_endpoint` decorator? Not sure.

Reply:
Hmm yeah, you mean just adding that as a decorator on all of our bouncer endpoints in general?

Though these have a kinda different subdomain check that happens in the authentication layer in `validate_remote_server`:
```
    if (
        get_subdomain(request) != Realm.SUBDOMAIN_FOR_ROOT_DOMAIN
        and not settings.DEVELOPMENT_DISABLE_PUSH_BOUNCER_DOMAIN_CHECK
    ):
        # Sometimes we may want to test push bouncer logic in development.
        raise JsonableError(_("Invalid subdomain for push notifications bouncer"))
```

Which is a fairly different treatment since we have the bouncer subdomain in `ROOT_SUBDOMAIN_ALIASES` I think? And then we do the check above. So this sounds like it would be a bit of a rework of this current way of doing this.

Reply:
Oh yeah, that's probably fine.